include/mra/kernels/multiply.h

```diff
@@ -31,6 +31,12 @@ namespace mra {
       Key<NDIM> key,
       size_type K)
     {
+      // ### Draft
+      // compare the keys of nodeA and nodeB to determine difference in level say n;
```

Comment:
No need to compare them, just use the key that provided as input to determine the desired level.

Reply:
The interface needs update. Both nodes need at least one extra level of refinement.

Reply:
Yes, but the key identifies the target node to which we will compress up.

Reply:
So the `task` does the comparison and sends the final key?

Reply:
Yes. Thinking about it, we need to pass in the keys of the inputs as well (so we can call fcube_for_mul). So yes, the interface needs to change...